### PR TITLE
Tidy up the backwards pass of type 1

### PR DESCRIPTION
**Pull request by WardBrian**
1. Clean up something where we were doing an fftshift, multiply, and ifftshift to just do 1 fftshift of the thing we were multiplying by
2. Remove the need to call FINUFFT multiple times in the points derivative by using the built-in batching functionality
3. Remove the numpy logic creating the ramp and replace it with a torch-native solution

### CODE REVIEW

File: pytorch_finufft/functional.py [hunk 1]
```diff
@@ -799,55 +799,39 @@ def backward(
...
-                    isign=_i_sign,
-                    **finufftkwargs,
-                )
-
-                grad_points = (backprop_ramp.conj() * values).real
-
-                grads_points.append(grad_points)
-
-            grads_points = torch.stack(grads_points)
+            ).to(device)
```
Comment (eickenberg) [inline pytorch_finufft/functional.py:823]: needed if the `torch.arange` is already on `device`? I think it shouldn't be necessary

File: pytorch_finufft/functional.py [hunk 2]
```diff
@@ -799,55 +799,39 @@ def backward(
...
+
+            # we can't batch in 1d case so we squeeze and fix up the ouput later
+            ramped_grad_output = (
+                coord_ramps * grad_output[np.newaxis] * 1j * _i_sign
+            ).squeeze()
```
Comment (eickenberg) [inline pytorch_finufft/functional.py:828]: is this because the coord ramps don't prepend a dimension when cartesian product is called with only 1 array?

File: pytorch_finufft/functional.py [hunk 3]
```diff
@@ -799,55 +799,39 @@ def backward(
...
+            backprop_ramp = nufft_func(
+                *points, ramped_grad_output, isign=_i_sign, **finufftkwargs
+            )
+            grads_points = torch.atleast_2d((backprop_ramp.conj() * values).real)
```
Comment (eickenberg) [inline pytorch_finufft/functional.py:832]: if the points were given as a strict 1D array, we probably need to give a gradient of the same shape back, because otherwise these might become hard to in-place add together

**Review by eickenberg — APPROVED** [review]
very very cool. some minor questions

File: pytorch_finufft/functional.py (hunk 3 above)
Comment (WardBrian) [reply pytorch_finufft/functional.py:832]: You're correct, but also it seems like at the moment the code doesn't handle 1-d points tensors at all. We need to fix this in #73

File: pytorch_finufft/functional.py (hunk 2 above)
Comment (WardBrian) [reply pytorch_finufft/functional.py:828]: The squeeze is because finufft doesn't support a batching dimension of length 1: https://github.com/flatironinstitute/finufft/blob/604e91fb45f95a867a79136a72c550305db9c1e5/python/finufft/finufft/_interfaces.py#L457C12-L459 (arguably a finufft issue)
